### PR TITLE
Create Application API endpoints

### PR DESCRIPTION
**Pull request by mwyld**
Initial structure to be built upon, this should satisfy the initial create application page.

- I have separated the GEF api tests into there own folder and ran coverage on this only to speed up writing tests
- I have commented out a couple of console logs that provided little help (I always had to scroll to see api test result)
- I have left pagination code in, which works incase this is needed in the future

### CODE REVIEW

File: portal-api/api-test-setup.jest.config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ mockFiles.forEach((mockFile) => {
   jest.mock(mockFile);
 });
 
-console.log(`MOCKED FILES: \n${mockFiles.join('\n')}`);
+// console.log(`MOCKED FILES: \n${mockFiles.join('\n')}`);
```
Comment (ttbarnes) [inline portal-api/api-test-setup.jest.config.js:16]: I think we could just remove these console.logs?

File: portal-api/api-tests/v1/gef/application.api-test.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      await as(aMaker).post(allItems[14]).to(baseUrl);
+      await as(aMaker).post(allItems[15]).to(baseUrl);
+
+
+      // MW: couldn't get the promise.all running in sequential order
+      // await allItems.map(async (item) => {
+      //   return as(aMaker).post(item).to(baseUrl);
+      // })
+
+      // await Promise.all(promise);
```
Comment (ttbarnes) [inline portal-api/api-tests/v1/gef/application.api-test.js:63]: remove?

**Review by ttbarnes — APPROVED** [review]
LGTM, probably worth removing the commented out console logs

File: portal-api/src/v1/gef/routes.js [hunk 3]
```diff
@@ -1,34 +1,31 @@
 const express = require('express');
 const { validate } = require('../../role-validator');
 const mandatoryCriteriaVersioned = require('./controllers/mandatoryCriteriaVersioned.controller');
+const application = require('./controllers/application.controller');
 
 const router = express.Router();
 
+// Application
+router.route('/application')
+  .get(application.getAll)
```
Comment (pricea1) [inline portal-api/src/v1/gef/routes.js:10]: Can we add a user validation on the GET requests too
e.g.  .get(
    validate({ role: ['maker', 'checker'] }),
    deals.findOne,
  )

**Review by pricea1 — APPROVED** [review]
Just the comment about adding validation to the GET otherwise LFTM

File: portal-api/api-test-setup.jest.config.js (hunk 1 above)
Comment (mwyld) [reply portal-api/api-test-setup.jest.config.js]: console.log removed

File: portal-api/api-tests/v1/gef/application.api-test.js (hunk 2 above)
Comment (mwyld) [reply portal-api/api-tests/v1/gef/application.api-test.js:63]: I have left this in as I may use it to refactor the code slightly, I couldn't get it working at the time

File: portal-api/src/v1/gef/routes.js (hunk 3 above)
Comment (mwyld) [reply portal-api/src/v1/gef/routes.js]: updated

**Comment by mwyld** [comment]
ready to be re-reveiwed

**Review by ttbarnes — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**